pinot-core/src/main/java/com/linkedin/pinot/core/data/manager/realtime/HLRealtimeSegmentDataManager.java

```diff
@@ -165,12 +165,12 @@ public HLRealtimeSegmentDataManager(final RealtimeSegmentZKMetadata realtimeSegm
       resourceTmpDir.mkdirs();
     }
     // create and init stream level consumer
-    _streamMetadata = new StreamMetadata(tableConfig.getIndexingConfig().getStreamConfigs());
-    _streamConsumerFactory = StreamConsumerFactoryProvider.create(_streamMetadata);
+    _streamConfig = new StreamConfig(tableConfig.getIndexingConfig().getStreamConfigs());
+    _streamConsumerFactory = StreamConsumerFactoryProvider.create(_streamConfig);
     _streamLevelConsumer = _streamConsumerFactory.createStreamLevelConsumer(
-        HLRealtimeSegmentDataManager.class.getSimpleName() + "-" + _streamMetadata.getKafkaTopicName());
+        HLRealtimeSegmentDataManager.class.getSimpleName() + "-" + _streamConfig.getKafkaTopicName());
```

Comment:
Note to change getKafkaTopicName() to a more generic name

Reply:
Will take care of this in a following PR, which will 
1. Unify the StreamConfig and StreamProviderConfig, so that we have only a single config object moving around
2. Split up the methods in StreamConfig, keeping only generic ones in the base class. All specific ones will go into StreamConfig wrappers specific to the stream implementation